src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -862,18 +1155,49 @@ def list_environment_locations(cmd):
     return res_locations
 
 
-def check_env_name_on_rg(cmd, env, resource_group_name, location):
+def check_env_name_on_rg(cmd, env, resource_group_name, location, custom_location_id, connected_cluster_id):
+    env_dict = parse_resource_id(env)
+    env_name = env_dict.get("name")
+    resource_type = env_dict.get("resource_type")
+    if resource_type is None:
+        if custom_location_id or connected_cluster_id:
+            resource_type = CONNECTED_ENVIRONMENT_TYPE
```

Comment:
Format the resource type here, instead to call Tolower() everywhere, like line 1171 and or line 1182